module/applications/components/inventory.mjs

```diff
@@ -47,268 +269,227 @@ export default class InventoryElement extends HTMLElement {
-   * TODO: Remove filtering code from dnd5e-inventory when all sheets use item-list-controls.
-   * Apply the current set of filters to the inventory list.
-   * @param {FilterState5e} state  The filter state to apply.
-   * @protected
+   * Cache section data to avoid expensive lookups during resize events.
+   * @internal
    */
-  _applyFilters(state) {
-    let items = this.app._filterItems?.(this.document.object.items, state.properties);
-    if ( !items ) return;
-    const elementMap = {};
-    this.querySelectorAll(".inventory-list .item-list .item").forEach(el => {
-      elementMap[el.dataset.itemId] = el;
-      el.hidden = true;
+  _cacheSections() {
+    this.#sections = Array.from(this.querySelectorAll(".items-section")).map(section => {
+      const minWidth = Number(section.dataset.columnMinWidth);
+      const descriptor = { element: section };
+      if ( Number.isFinite(minWidth) ) descriptor.minWidth = minWidth;
+      const columns = Array.from(section.querySelectorAll(".items-header [data-column-id]")).reduce((obj, el) => {
+        const { columnId: id, columnWidth: width, columnPriority: priority } = el.dataset;
+        obj[id] = { id, width: Number(width), priority: Number(priority) };
+        return obj;
+      }, {});
+      // Descending order of priority.
+      descriptor.columns = Object.values(columns).sort((a, b) => b.priority - a.priority);
+      const elements = section.querySelectorAll(":is(.item-row, .items-header) [data-column-id]");
+      descriptor.elements = Array.from(elements).reduce((obj, el) => {
+        const { columnId: id } = el.dataset;
+        const els = obj[id] ??= [];
+        els.push(el);
+        return obj;
+      }, {});
+      return descriptor;
     });
-    for ( const item of items ) {
-      const el = elementMap[item.id];
-      if ( el ) el.hidden = false;
-    }
   }
 
   /* -------------------------------------------- */
-  /*  Properties                                  */
+  /*  Event Listeners & Handlers                  */
   /* -------------------------------------------- */
 
   /**
-   * Reference to the application that contains this component.
-   * @type {Application}
-   */
-  #app;
-
-  /**
-   * Reference to the application that contains this component.
-   * @type {Application}
+   * Prepare an array of context menu options which are available for inventory items.
+   * @param {Item5e} item          The item.
+   * @param {HTMLElement} element  The item's rendered element.
+   * @returns {ContextMenuEntry[]}
    * @protected
    */
-  get app() { return this.#app; }
+  _getContextOptions(item, element) {
+    // Standard options.
+    const options = [{
+      name: "DND5E.ItemView",
+      icon: '<i class="fa-solid fa-eye fa-fw"></i>',
+      callback: li => this._onAction(li, "view")
+    }, {
+      name: "DND5E.ContextMenuActionEdit",
+      icon: '<i class="fa-solid fa-edit fa-fw"></i>',
+      condition: () => item.isOwner && !item.collection.locked,
```

Comment:
`item.collection.locked` doesn't evaluate to `true` in a locked compendium, this will probably need to check `item.collection.locked || item.actor.collection.locked` to ensure it handles locked compendiums and actors with only observer permissions.